### ISSUE TITLE
Feature request: access to other fields in the bibtex entry

### ISSUE THREAD

**Issue by jdanish** [inukshuk/jekyll-scholar#214]
Hello. I have a feature request (or question if it already exists!!). I want to have a repository in my install, but I want to be able to set custom file names so that they reflect the content better. I tried setting up a custom field for this, but cannot access it in my bib.html, or at least this seems to not be possible. Is there an easy way to do this and / or would this be a possible future feature?  Thanks!

**Comment by jdanish** [comment]
Also thanks for the great plugin!

**Comment by inukshuk** [comment]
What kind of filenames do you want to use?

At the moment the file repository works by indexing files based on BibTeX keys; the keys are under your control (with some constraints) so they are already custom in the sense that you can change them to reflect your desired file names. Also, it would be easy to adjust the code to make it possible to use a different field, not the key, as the file name (the 'delimiter' can already be configured). Is that what you're looking for? An example would help.

Alternatively, you can write a custom template and create the links yourself (based on a custom field). All custom fields should be available in the template via `entry.x`, with `x` being the name of your custom field.

**Comment by jdanish** [comment]
Thanks for your reply!

The reason I ran into this issue is that EndNote (the reference manager I use) automatically creates the key for me and I have no control over it. So I have keys that read R1345 and what I want is files that are named something like 2017-Author-Title-Here.pdf.

It turns out I was using a custom field named fileName (mixed case), which was not working. Given your comment I tried filename (all lower) and now it works just fine. 

So, I can now do what I want! No edit needed. Thanks!!

Joshua
